src/lib/pages/instantiate/instantiate.tsx

```diff
@@ -238,19 +242,28 @@ const Instantiate = ({ onComplete }: InstantiatePageProps) => {
           label: msgObject.label,
           adminAddress: msgObject.admin,
           initMsg: JSON.stringify(msgObject.msg, null, 2),
-          assets:
-            msgObject.funds.length === 0
-              ? [{ denom: "", amount: "" }]
-              : msgObject.funds.map((fund) => ({
-                  ...fund,
-                  amount: demicrofy(fund.amount as U<Token>).toFixed(),
-                })),
         });
+
+        if (msgObject.funds.length) {
+          resetAssets({
+            assetsSelect: [{ denom: "", amount: "" }],
+            assetsJson: jsonPrettify(JSON.stringify(msgObject.funds)),
+            attachFundOption: AttachFundsType.ATTACH_FUNDS_JSON,
```

Comment:
be aware that using reset changes the form default values to the one you provide